flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-iceberg/src/main/java/org/apache/flink/cdc/connectors/iceberg/sink/utils/IcebergTypeUtils.java

```diff
@@ -0,0 +1,214 @@
+                break;
+            case BOOLEAN:
+                fieldGetter = row -> row.getBoolean(fieldPos);
+                break;
+            case BINARY:
+            case VARBINARY:
+                fieldGetter = row -> row.getBinary(fieldPos);
+                break;
+            case DECIMAL:
+                final int decimalPrecision = DataTypeChecks.getPrecision(fieldType);
+                final int decimalScale = DataTypeChecks.getScale(fieldType);
+                fieldGetter =
+                        row -> {
+                            DecimalData decimalData =
+                                    row.getDecimal(fieldPos, decimalPrecision, decimalScale);
+                            return org.apache.flink.table.data.DecimalData.fromBigDecimal(
+                                    decimalData.toBigDecimal(), decimalPrecision, decimalScale);
+                        };
+                break;
+            case TINYINT:
+                fieldGetter = row -> row.getByte(fieldPos);
+                break;
+            case SMALLINT:
+                fieldGetter = row -> row.getShort(fieldPos);
+                break;
+            case BIGINT:
+                fieldGetter = row -> row.getLong(fieldPos);
+                break;
+            case FLOAT:
+                fieldGetter = row -> row.getFloat(fieldPos);
+                break;
+            case DOUBLE:
+                fieldGetter = row -> row.getDouble(fieldPos);
+                break;
+            case INTEGER:
+            case DATE:
+                fieldGetter = row -> row.getInt(fieldPos);
+                break;
+            case TIME_WITHOUT_TIME_ZONE:
+                fieldGetter =
+                        (row) -> {
+                            LocalDateTime localDateTime =
+                                    row.getTimestamp(
+                                                    fieldPos,
+                                                    DataTypeChecks.getPrecision(fieldType))
+                                            .toLocalDateTime();
+                            return DateTimeUtil.microsFromTimestamp(localDateTime);
+                        };
+                break;
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+                fieldGetter =
+                        (row) -> {
+                            Instant instant =
+                                    row.getLocalZonedTimestampData(
+                                                    fieldPos,
+                                                    DataTypeChecks.getPrecision(fieldType))
+                                            .toInstant();
+                            return DateTimeUtil.microsFromTimestamptz(
+                                    OffsetDateTime.ofInstant(instant, zoneId));
+                        };
+                break;
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+            case TIMESTAMP_WITH_TIME_ZONE:
```

Comment:
Is it right that `TIMESTAMP_WITH_LOCAL_TIME_ZONE` and `TIMESTAMP_WITH_TIME_ZONE` have the same code?